search-provider/eks-discovery-feed-provider.c

```diff
@@ -181,19 +182,44 @@ underscorify (const gchar *string)
 }
 
 static void
+add_key_value_pair_to_variant (GVariantBuilder *builder,
+                               const char      *key,
+                               const char      *value)
+{
+  g_variant_builder_open (builder, G_VARIANT_TYPE ("{ss}"));
+  g_variant_builder_add (builder, "s", key);
+  g_variant_builder_add (builder, "s", value);
+  g_variant_builder_close (builder);
+}
+
+static void
 add_key_value_pair_from_model_to_variant (EkncContentObjectModel *model,
                                           GVariantBuilder        *builder,
                                           const char             *key)
 {
-  g_variant_builder_open (builder, G_VARIANT_TYPE ("{ss}"));
   g_autofree gchar *value = NULL;
   g_autofree gchar *underscore_key = underscorify (key);
   g_object_get (model, key, &value, NULL);
-  g_variant_builder_add (builder, "s", underscore_key);
-  g_variant_builder_add (builder, "s", value);
-  g_variant_builder_close (builder);
+  add_key_value_pair_to_variant (builder, underscore_key, value);
+}
+
+static gchar *
+select_random_string_from_variant(GVariant *variant)
+{
+  gsize size = g_variant_n_children(variant);
+  gint index = g_random_int_range (0, size);
+  /* Note that the extracted child_variant here is of type 'v', we still
+   * need to unwrap it one more time to get the string */
```

Comment:
I wasn't able to get this to work :( I tried using something like `g_variant_get_child(variant, index, "(s)", &str, NULL)`, but GVariant was complaining that the underlying value is of type `v` and wouldn't unbox the inner string.